### ISSUE TITLE
Построение OpenCV из исходных кодов

### ISSUE THREAD

**Issue by valentina-kustikova** [itlab-vision/itlab-vision#76]
Необходимо обеспечить на Travis-CI скачивание исходных кодов OpenCV (версия 2.4.9) и их сборку вместо установки пакета.

**Comment by valentina-kustikova** [comment]
@kirill-kornyakov пришли, пожалуйста, пример.

**Comment by kirill-korniakov** [comment]
Воть: https://github.com/Itseez/opencv_contrib/blob/master/.travis.yml Только конечно к нему нужно отнестись творчески.

**Comment by KruchDmitriy** [comment]
Добрый день.

Скажите пожалуйста, make -j5 означает, что make может одновременно
обрабатывать до 5 задач? Если да, то почему именно 5?

С уважением,
Кручинин Д.

2014-05-05 20:09 GMT+04:00 Kirill Kornyakov notifications@github.com:

**Comment by kirill-korniakov** [comment]
Этот параметр указывает на то, сколько потоков использовать. Чаще всего сейчас на машинах 4 ядра, поэтому 5 (4 + 1) потоков. Все это условно, так что сильно заморачиваться не стоит, но есть мнение что так оптимальнее всего (число ядер + 1).

**Comment by KruchDmitriy** [comment]
Спасибо)

У меня есть пробная версия файла .travis.yml в pull request'е, посмотрите
пожалуйста.
Возникает ошибка:
unzip: cannot find or open download.zip, download.zip.zip or
download.zip.ZIP
вот ссылка на построение
https://travis-ci.org/UNN-VMK-Software/itlab-vision/jobs/24791764

9 мая 2014 г., 17:35 пользователь Kirill Kornyakov <notifications@github.com

**Comment by KruchDmitriy** [comment]
Забыл сказать, pull request - Build OpenCV from source files.

9 мая 2014 г., 17:45 пользователь Дмитрий Кручинин
kruch.dmitriy@gmail.comнаписал:

**Comment by DolotovEvgeniy** [comment]
так вроде по этой ссылке скачивается файл opencv-2.4.9.zip, а не
download.zip.

9 мая 2014 г., 17:48 пользователь Дмитрий notifications@github.comнаписал:

**Comment by KruchDmitriy** [comment]
команда wget сохраняет его с именем download

**Comment by druzhkov-paul** [comment]
По-моему, вы сами ответили на свой вопрос. =) Команда `wget` сохраняет этот файл с именем `download`, а для распаковки указано имя `download.zip`. Следовательно, должно быть либо `unzip download`, либо в команде `wget` можно явно указать имя результирующего файла с помощью опции `-O`.

**Comment by KruchDmitriy** [comment]
Добрый вечер.

Возникла проблема во время построения opencv, которую я не могу решить
самостоятельно. Прошу посмотреть ход сборки
https://travis-ci.org/UNN-VMK-Software/itlab-vision/jobs/24990656 .

11 мая 2014 г., 21:57 пользователь Pavel Druzhkov
notifications@github.comнаписал:

**Comment by KruchDmitriy** [comment]
Ошибка возникла с clang. С gcc всё до конца собралось и отработало как надо
(кроме теста фильтра antique).
https://travis-ci.org/UNN-VMK-Software/itlab-vision/jobs/24990656

12 мая 2014 г., 20:23 пользователь Дмитрий Кручинин <kruch.dmitriy@gmail.com

**Comment by druzhkov-paul** [comment]
@KruchDmitriy , не обязательно строить всю библиотеку OpenCV, кажется, достаточно модулей `core`, `imgproc`, `highgui` и `ts`. Они собираются нормально. К тому же построение должно стать гораздо быстрее.

**Comment by KruchDmitriy** [comment]
Заменил. Проблема не решена.

**Comment by druzhkov-paul** [comment]
Да, точно, `ts` как раз и не собирается. Попробуйте собирать OpenCV с помощью gcc, а photoeffects clang'ом.
